sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1167,6 +1192,74 @@ Exceptions:
   `property::graph::updatable` was not set when the executable graph was created.
 * Throws with error code `invalid` if any node in `nodes` is not part of the
   graph.
+
+|
+[source, c++]
+----
+void
+update(const command_graph<graph_state::modifiable>& graph);
+----
+
+|Updates all of the nodes in the executable graph with parameters from a
+topologically identical modifiable graph. A topologically identical graph is one
+with the same structure of nodes and edges. Additionally, all nodes and edges
```

Comment:
The phrase "topologically identical" seems pretty clear when we consider nodes and internal edges.  What about cases where a node depends on an external event?  What does "topologically identical" mean in that case?

What if the application creates an output event from a graph by calling `command_graph::get_event`?  Does this affect the topology?

Also, how does whole graph update work when there are external events?  Are external events in the "target" graph updated with the equivalent external events in the "source" graph?

Reply:
> The phrase "topologically identical" seems pretty clear when we consider nodes and internal edges. What about cases where a node depends on an external event? What does "topologically identical" mean in that case?

As the dynamic events proposal stands these external events do not create edges in the graph, so they currently don't fall under this definition. Probably it would make sense to additionally mandate that nodes must have the same number of external dependencies.

> What if the application creates an output event from a graph by calling `command_graph::get_event`? Does this affect the topology?

As above I would say it does not, and I don't think we need to mandate anything here.
 
> Also, how does whole graph update work when there are external events? Are external events in the "target" graph updated with the equivalent external events in the "source" graph?

I think that would be the intention yes, providing that the number of external dependencies match.

Reply:
Your proposed interpretation sounds reasonable.  However, I don't think the current spec wording is clear about whether external events are considered when determining whether two graphs are "topologically identical".  I think you are proposing that in order for two graphs to be topologically identical, corresponding nodes must have the same number of "input" external events and the same number of "output" external events.  I think this is fine, but the spec wording should be changed to make this clear.

Reply:
I've added some wording for this to the definition of topologically identical, however I'm a bit hesitant to add any more since that proposed specification is still in flight and I would rather not have PRs depending on each other. Any inconsistencies I think can be addressed in whichever PR is not merged first.